Feat/user admin

Mani-sub opened the pull request:
- Pull the branch into local for testing.
- run `yarn install` than `yarn run serve`
- On the UI - click on Sign In button from the header view.
- Use the credentials - "username - admin", "password - JW24G.xR" and login. 

- The admin panel can be viewed similar to that of prototype.

joschrew commented:
I don't have much experience with front-end development. Did you change the package manager from yarn to npm? Can both be used interchangeably or in parallel, or did you just get that mixed up?

Mani-sub commented:
> 

I think i got mixed up while adding new packages to the repo and installing the dependencies. Will remove the package-lock.json file and update the PR. Continue using yarn instead of npm for better performance of installing larger package files.

joschrew commented:
ok, thanks for explaining and the changes.
btw: I started reviewing this pull-request, but it will take time. Previously I thought it would be better if you just push directly to develop, but I changed my mind. I think it is a good idea for me to review this because I need to learn more about the front-end.

Mani-sub commented:
> ok, thanks for explaining and the changes. btw: I started reviewing this pull-request, but it will take time. Previously I thought it would be better if you just push directly to develop, but I changed my mind. I think it is a good idea for me to review this because I need to learn more about the front-end.

Sure, just for an overview - the new feature changes related to Sign In button were in `src/layout/navigation-menu -> index.vue`, and in components module `login` and `modal` were added new. Apart from that, all the other components are added from the admin repo. No logical changes in here.

joschrew commented:
Ok, I am through with looking at this pull request. I am a impressed that you could move all code from the previous admin-frontend to the user-frontend without problems. Here are some things that you could maybe change:
- src/axios-config.ts: you set baseURL to `http://ola-hd.sub.uni-goettingen.de/api/`. I think you did this for testing but forgot to remove before committing. Btw, that is the "old" Prototype backend, maybe our dev-server (http://141.5.99.53/api/) or staging-server (http://141.5.104.244/api) would be better, but currently no big difference. 
- src/router/index.ts: route dashview (Line 63). 3 child-components are set. One has path "", name "home" and component "Dashboard". I don't know much about vue and everything but maybe this has to be `Home` as a Component like in line 20? And I think I get warn-messages about this when using dev-tools, but I'm not sure.
- After login, I would prefer the dashboard to be shown without the search on top, but I think that is a decision of PO
- In the table of the dasboard the PID of an uploaded archive is a link. Previously in the Dashboard, this fact was a bit more visible from my point of view, because the PID was in blue other than the other columns. Maybe you can make it more visible (blue?) that this PID can be clicked like previously.

Mani-sub commented:
> Ok, I am through with looking at this pull request. I am impressed that you could move all code from the previous admin-frontend to the user-frontend without problems. Here are some things that you could maybe change:
> 
> * src/axios-config.ts: you set baseURL to `http://ola-hd.sub.uni-goettingen.de/api/`. I think you did this for testing but forgot to remove before committing. Btw, that is the "old" Prototype backend, maybe our dev-server (http://141.5.99.53/api/) or staging-server (http://141.5.104.244/api) would be better, but currently no big difference.
> * src/router/index.ts: route dashview (Line 63). 3 child-components are set. One has path "", name "home" and component "Dashboard". I don't know much about vue and everything but maybe this has to be `Home` as a Component like in line 20? And I think I get warn-messages about this when using dev-tools, but I'm not sure.
> * After login, I would prefer the dashboard to be shown without the search on top, but I think that is a decision of PO
> * In the table of the dasboard the PID of an uploaded archive is a link. Previously in the Dashboard, this fact was a bit more visible from my point of view, because the PID was in blue other than the other columns. Maybe you can make it more visible (blue?) that this PID can be clicked like previously.

- fixed warning error of routing in console dev tools, changed pid text color to use blue and changed base url in axios config file. 
- The search on top is a part of the current user story Search Implementation, so once this is added, than we can view the dashboard without search on top.

joschrew commented on src/axios-config.ts:
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios';
+
+const instance = axios.create({
+    baseURL: 'http://141.5.99.53/api/',
```
```suggestion
    baseURL: '/api/',
```

joschrew commented:
Ok I testet the changes LGTM now execept the thing about `src/axios-config.ts`, sry for possible confusing. Next try of explanation:
What a said was just about the local development environment. If you want to use the frontend for development and start it with yarn on your own machine, `/api` will not be available so you insert what you did into the src/axios-config.ts as a workaround to be able to test how everything works in conjunction with the backend. But when we deploy the whole programm (backend + frontend) on our vms we need the code that i posted there.
About search, sry, I maybe dimly rember you already said that in one of our meetings.

Mani-sub commented:
> Ok I testet the changes LGTM now except the thing about `src/axios-config.ts`, sry for possible confusing. Next try of explanation: What a said was just about the local development environment. If you want to use the frontend for development and start it with yarn on your own machine, `/api` will not be available so you insert what you did into the src/axios-config.ts as a workaround to be able to test how everything works in conjunction with the backend. But when we deploy the whole programm (backend + frontend) on our vms we need the code that i posted there. About search, sry, I maybe dimly rember you already said that in one of our meetings.

Yes, for frontend testing purpose i have used the dev-server url. I have changed to use `/api/` and updated the PR.
